scripts/linux/smokeTestHelper.sh

```diff
@@ -0,0 +1,220 @@
+  echo "[PASS] The current artifact size ($incomingSize) is withing the range of $minAcceptingArtifactSize Byte to $maxAcceptingArtifactSize Byte" 
+}
+
+check-version-content-difference()
+{
+  # Compare if the content of two files are difference. 
+  #   Return true - if the two files are different
+  #   Return false - if the two files are byte-exact identical.
+  #
+  # $1 - path to file1
+  # $2 - path to file2
+
+  # Compare the previous and current artifact content diff (by byte)
+  if cmp --silent $1 $2; then
+    echo "[FAIL] The current ($1) and previous ($2) artifacts are identical."
+    exit 1;
+  fi
+  echo "[PASS] The current and previous artifact version have a different content"
+}
+
+check-github-pmc-artifacts-similarity()
+{
+  # Compare the artifact if github and pmc are the same artifact
+  #  Return true - if the artifacts from two sources are identical
+  #  Return false - if the artifacts from two sources are NOT identical
+  #
+  # $1 - path to PMC artifact
+  # $2 - path to Github artifact directory
+  # $3 - package name
+
+  echo ""
+  ghArtifact="$2/$3_$gitHubArtifactSuffix"
+  echo "The artifact from GitHub: $ghArtifact"
+  [[ -f "$ghArtifact" ]] || { echo "[FAIL] The artifact from GitHub doesn't exist ($ghArtifact)"; exit 1; }
+  if cmp --silent $1 $ghArtifact; then
+    echo "[PASS] The artifact content in PMC and the GitHub are identical."
+  else
+    echo "[FAIL] The artifact content in PMC and the GitHub are NOT identical."
+    exit 1;
+  fi
+}
+
+download-artifact-from-pmc-apt()
+{
+  # Download artifact for a package named ($1) using uri acquired by `apt` to a location ($2).
+  # Then suffix the name of the artifact with "_pmc".
+  #
+  # $1 - downloading an artifact for a package name (latest by default)
+  #    Note: To download a specific version, the ($1) can be specified as "aziot-edge=1.2.10-1"
+  # $2 - target output directory
+
+  pkgName=${1%=*}
+  uri=$(apt-get install --reinstall --print-uris -qq $1 | cut -d"'" -f2 | grep "/$pkgName/")
+  [[ -z "$uri" ]] && { echo "[FAIL] Package ($1) cannot be found in a known linux repository"; exit 1;}
+  artifactName=${uri##*/}
+  targetArtifactPath="$(realpath $2)/${artifactName%*.deb}_pmc.deb"
+  
+  cmd="wget -q $uri -O $targetArtifactPath"
+  echo "Running: $cmd"
+  $cmd
+}
+
+wait-for-dpkg-lock()
+{
+  # Wait for task 'apt.systemd.daily' to be completed. This is helpful if you later
+  # need to use `apt`. 
+  #
+  # $1 - wait time in seconds
+
+  waitAttempts=$1
+  sleepDuration=10
+  for (( i=0; $i<$waitAttempts; i++ ))
+  do
+    if [[ $(ps aux | grep -c apt.systemd.daily) == 1 ]]; then
+      break
+    fi
+    sleep $sleepDuration
+  done
+}
+
+setup-focal-source-apt()
+{
+  # 1ES image for Ubuntu20.04 (Focal) actually has a misconfigured apt repo source
```

Comment:
Can it be fixed in 1ES?

Reply:
We pull the 1ES Base image straight from their Focal.  But it still show the bionic apt. Not sure if there is a way to complain to them directly. This workaround works well in the mean time.